### PR TITLE
track more things in google analytics #190

### PR DESCRIPTION
**Pull request by kdalkafoukis**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -175,6 +178,8 @@ class App extends Component {
   };
 
   toggleLeftDrawer = (isItOpen) => () => {
+    const pathname = this.props.history.location.pathname;
+    reactGAmodalview(pathname);
```
Comment (sebastianovide) [inline src/App.js]: what about if isItOpen is false ? are you still tracking a page view ?

File: src/components/Map.js [hunk 2]
```diff
@@ -177,7 +179,10 @@ class Map extends Component {
         el.className = 'marker';
         el.id = feature.properties.id;
         el.style.backgroundImage = `url(${feature.properties.thumbnail}), url(${placeholderImage}) `;
-        el.addEventListener('click', () => this.setState({openDialog:true,feature}));
+        el.addEventListener('click', () => {
+          reactGAevent('User','Photo in the map');
```
Comment (sebastianovide) [inline src/components/Map.js]: how is "Photo in the map" part of the "User" category ?

File: src/components/PhotoPage.js [hunk 3]
```diff
@@ -89,7 +90,8 @@ class PhotoPage extends Component {
         try {
           const res = await dbFirebase.uploadPhoto(data);
           console.log(res);
-          this.openDialog("Photo was uploaded successfully", this.handleGoBackButton);
+          reactGAevent('Photo Page','Photo Uploaded');
```
Comment (sebastianovide) [inline src/components/PhotoPage.js]: is "Photo Page" an event category ?

File: src/functions/googleAnalytics.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import ReactGA from 'react-ga';
```
Comment (sebastianovide) [inline src/functions/googleAnalytics.js]: what's the added value of this file to RectGA.... the functions looks exactly the same

File: src/App.js [hunk 5]
```diff
@@ -19,6 +18,7 @@ import Login from './components/Login';
 import AboutPage from './components/AboutPage';
 import TutorialPage from './components/TutorialPage';
 import DrawerContainer from './components/DrawerContainer';
+import { reactGAevent,reactGAmodalview, reactGApageview } from './functions/googleAnalytics';
```
[the remark this answers is not in the document]
Comment (kdalkafoukis) [reply src/App.js]: removed

File: src/App.js [hunk 6]
```diff
@@ -68,6 +68,7 @@ class App extends Component {
         });
 
       }, error => {
+        reactGAevent("User",'Denied to enable Location');
```
[the remark this answers is not in the document]
Comment (kdalkafoukis) [reply src/App.js]: removed

File: src/App.js (hunk 1 above)
Comment (kdalkafoukis) [reply src/App.js]: fixed

File: src/components/Map.js (hunk 2 above)
Comment (kdalkafoukis) [reply src/components/Map.js]: changed

File: src/components/PhotoPage.js (hunk 3 above)
Comment (kdalkafoukis) [reply src/components/PhotoPage.js]: changed to Photo, yes it was a category

File: src/functions/googleAnalytics.js (hunk 4 above)
Comment (kdalkafoukis) [reply src/functions/googleAnalytics.js]: removed

File: src/components/Map.js [hunk 7]
```diff
@@ -178,7 +180,10 @@ class Map extends Component {
         el.className = 'marker';
         el.id = feature.properties.id;
         el.style.backgroundImage = `url(${feature.properties.thumbnail}), url(${placeholderImage}) `;
-        el.addEventListener('click', () => this.setState({openDialog:true,feature}));
+        el.addEventListener('click', () => {
+          ReactGA.event('Map','User opened photo on the map');
```
Comment (sebastianovide) [inline src/components/Map.js]: which photo ?

File: src/components/Map.js [hunk 8]
```diff
@@ -151,7 +152,8 @@ class Map extends Component {
     });
   }
 
-  flyToGpsLocation = () =>{
+  flyToGpsLocation = () => {
+    ReactGA.event('Map','User pressed location fab');
```
Comment (sebastianovide) [inline src/components/Map.js]: location and zoom level ?

File: src/App.js [hunk 9]
```diff
@@ -134,6 +136,7 @@ class App extends Component {
   };
 
   handlePhotoClick = () => {
+     ReactGA.event('User','photo modal opened');
```
Comment (sebastianovide) [inline src/App.js]: it is a photo event not a user event

File: src/App.js [hunk 10]
```diff
@@ -68,6 +68,7 @@ class App extends Component {
         });
 
       }, error => {
+        ReactGA.event("User",'Location Error');
```
Comment (sebastianovide) [inline src/App.js]: it is not a user event

File: src/components/Map.js (hunk 8 above)
Comment (kdalkafoukis) [reply src/components/Map.js]: Max Stephan said zoom level is not needed - location is already tracked by GA

File: src/App.js (hunk 9 above)
Comment (kdalkafoukis) [reply src/App.js]: fixed

Comment (kdalkafoukis) [reply src/App.js]: changed

File: src/App.js (hunk 10 above)
Comment (kdalkafoukis) [reply src/App.js]: changed

**Pull request closed without merging**